arrow-array/src/array/run_array.rs

```diff
@@ -233,6 +275,63 @@ impl<R: RunEndIndexType> RunArray<R> {
         }
         Ok(physical_indices)
     }
+
+    /// Returns a `RunArray` with zero offset and length matching the last value
+    /// in run_ends array.
+    pub fn into_non_sliced_array(self) -> Result<Self, ArrowError> {
```

Comment:
```suggestion
    pub fn materialize_array_offset(self) -> Result<Self, ArrowError> {
```

Or something, if we move this to be an implementation detail of IPC this effectively becomes an implementation detail